packages/main/src/TextArea.ts

```diff
@@ -0,0 +1,707 @@
+	maxlength?: number;
+
+	/**
+	 * Determines whether the characters exceeding the maximum allowed character count are visible
+	 * in the component.
+	 * <br><br>
+	 * If set to <code>false</code>, the user is not allowed to enter more characters than what is set in the
+	 * <code>maxlength</code> property.
+	 * If set to <code>true</code> the characters exceeding the <code>maxlength</code> value are selected on
+	 * paste and the counter below the component displays their number.
+	 *
+	 * @type {boolean}
+	 * @name sap.ui.webc.main.TextArea.prototype.showExceededText
+
+	 * @defaultvalue false
+	 * @public
+	 */
+	@property({ type: Boolean })
+	showExceededText!: boolean;
+
+	/**
+	 * Enables the component to automatically grow and shrink dynamically with its content.
+	 * <br><br>
+	 * @type {boolean}
+	 * @name sap.ui.webc.main.TextArea.prototype.growing
+	 * @defaultvalue false
+	 * @public
+	 */
+	@property({ type: Boolean })
+	growing!: boolean;
+
+	/**
+	 * Defines the maximum number of lines that the component can grow.
+	 *
+	 * @type {sap.ui.webc.base.types.Integer}
+	 * @name sap.ui.webc.main.TextArea.prototype.growingMaxLines
+	 * @defaultvalue 0
+	 * @public
+	 */
+	@property({ validator: Integer, defaultValue: 0 })
+	growingMaxLines!: number;
+
+	/**
+	 * Determines the name with which the component will be submitted in an HTML form.
+	 *
+	 * <br><br>
+	 * <b>Important:</b> For the <code>name</code> property to have effect, you must add the following import to your project:
+	 * <code>import "@ui5/webcomponents/dist/features/InputElementsFormSupport.js";</code>
+	 *
+	 * <br><br>
+	 * <b>Note:</b> When set, a native <code>input</code> HTML element
+	 * will be created inside the component so that it can be submitted as
+	 * part of an HTML form. Do not use this property unless you need to submit a form.
+	 *
+	 * @type {string}
+	 * @name sap.ui.webc.main.TextArea.prototype.name
+	 * @defaultvalue ""
+	 * @public
+	 */
+	@property()
+	name!: string;
+
+	/**
+	 * Defines the accessible ARIA name of the component.
+	 *
+	 * @type {string}
+	 * @name sap.ui.webc.main.TextArea.prototype.accessibleName
+	 * @public
+	 * @since 1.0.0-rc.15
+	 */
+	@property()
+	accessibleName!: string;
+
+	/**
+	 * Receives id(or many ids) of the elements that label the textarea.
+	 *
+	 * @type {string}
+	 * @name sap.ui.webc.main.TextArea.prototype.accessibleNameRef
+	 * @defaultvalue ""
+	 * @public
+	 * @since 1.0.0-rc.15
+	 */
+	@property()
+	accessibleNameRef!: string;
+
+	/**
+	 * @private
+	 */
+	@property({ type: Boolean })
+	focused!: boolean;
+
+	/**
+	 * @private
+	 */
+	 @property({ type: Boolean })
+		exceeding!: boolean;
+
+	/**
+	 * @private
+	 */
+	@property({ type: Object, multiple: true })
+	_mirrorText!: IndexedTokenizedText;
+
+	/**
+	 * @private
+	 */
+	@property({ noAttribute: true })
+	_maxHeight!: string;
+
+	/**
+	 * @private
+	 */
+	@property({ validator: Integer })
+	_width: number | undefined;
```

Comment:
`_width?: number` could be used here, or just set a defaultValue of 0 and use `_width!: number`